android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java

```diff
@@ -244,12 +248,29 @@ public void handleMessage(Message msg) {
                         }
                         msg = obtainMessage(SHOW_PROGRESS);
                         sendMessageDelayed(msg, Math.round(mProgressUpdateInterval));
+                        setFrame();
                     }
                     break;
             }
         }
     };
 
+    private void setFrame() {
+        if(frameQuality == null) return;
+        ((Runnable) () -> {
```

Comment:
I don't understand why you use a runnable here.
I think you want to do something asynchroniously. If that is the case, you should also use a Handler ?